src/hooks/executor_start.c

```diff
@@ -0,0 +1,76 @@
+#include <postgres.h>
+
+#include "executor_start.h"
+
+#include <executor/executor.h>
+#include <nodes/pg_list.h>
+#include <nodes/plannodes.h>
+
+#include "plan_tree_walker.h"
+#include "utils.h"
+
+typedef struct
+{
+    List *oidList;
+    bool  isIndexScan;
+} OperatorUsedCorrectlyContext;
+
+static bool operator_used_correctly_walker(Node *node, OperatorUsedCorrectlyContext *context)
+{
+    if(node == NULL) return false;
+    if(IsA(node, IndexScan)) {
+        context->isIndexScan = true;
```

Comment:
Can you explain what's happening here?
why do we set 'isIndexScan' to true, and then back to false on line 24

Reply:
It's traversing a tree. When it hits an IndexScan node, it's going to "go inside it". I'm noting that we're inside an index scan now. If we hit the Lantern operator inside an index scan, that's an acceptable way to use the operator. We return false. If we hit the Lantern operator and we are not inside an index scan, that is not an acceptable way to use the operator. The `if(list_member_oid(context->oidList, opExpr->opno) && !context->isIndexScan) {` will catch this, and return true.